rskj-core/src/test/java/co/rsk/peg/ReleaseTransactionBuilderTest.java

```diff
@@ -390,6 +388,86 @@ public void buildEmptyWalletTo_utxoProviderException() throws InsufficientMoneyE
+            tx.addInput(utxo3.getHash(), utxo3.getIndex(), mock(Script.class));
+
+            return null;
+        }).when(wallet).completeTx(any(SendRequest.class));
+
+        Optional<ReleaseTransactionBuilder.BuildResult> result = builder.buildBatchedPegouts(entries);
+
+        BtcTransaction tx = result.get().getBtcTx();
+        List<UTXO> selectedUTXOs = result.get().getSelectedUTXOs();
+
+        Assert.assertTrue(result.isPresent());
+
+        Assert.assertEquals(3, tx.getOutputs().size());
+        Assert.assertEquals(testEntry1.getAmount(), tx.getOutput(0).getValue());
+        Assert.assertEquals(testEntry2.getAmount(), tx.getOutput(1).getValue());
+        Assert.assertEquals(testEntry3.getAmount(), tx.getOutput(2).getValue());
+        Assert.assertEquals(testEntry1.getDestination(), tx.getOutput(0).getAddressFromP2PKHScript(NetworkParameters.fromID(NetworkParameters.ID_REGTEST)));
+        Assert.assertEquals(testEntry2.getDestination(), tx.getOutput(1).getAddressFromP2PKHScript(NetworkParameters.fromID(NetworkParameters.ID_REGTEST)));
+        Assert.assertEquals(testEntry3.getDestination(), tx.getOutput(2).getAddressFromP2PKHScript(NetworkParameters.fromID(NetworkParameters.ID_REGTEST)));
+
+        Assert.assertEquals(2, tx.getInputs().size());
+        Assert.assertEquals(utxo2.getHash(), tx.getInput(0).getOutpoint().getHash());
+        Assert.assertEquals(utxo2.getIndex(), tx.getInput(0).getOutpoint().getIndex());
+        Assert.assertEquals(utxo3.getHash(), tx.getInput(1).getOutpoint().getHash());
+        Assert.assertEquals(utxo3.getIndex(), tx.getInput(1).getOutpoint().getIndex());
+
+        Assert.assertEquals(2, selectedUTXOs.size());
+        Assert.assertEquals(utxo2.getHash(), selectedUTXOs.get(0).getHash());
+        Assert.assertEquals(utxo2.getIndex(), selectedUTXOs.get(0).getIndex());
+        Assert.assertEquals(utxo3.getHash(), selectedUTXOs.get(1).getHash());
+        Assert.assertEquals(utxo3.getIndex(), selectedUTXOs.get(1).getIndex());
+    }
+
+    @Test
+    public void test_BuildBatchedPegouts_insufficientMoneyException() throws InsufficientMoneyException {
+        Context btcContext = new Context(NetworkParameters.fromID(NetworkParameters.ID_REGTEST));
+
+        ReleaseRequestQueue.Entry testEntry1 = createTestEntry(123, 2);
+        ReleaseRequestQueue.Entry testEntry2 = createTestEntry(456, 4);
+        ReleaseRequestQueue.Entry testEntry3 = createTestEntry(789, 5);
+        List<ReleaseRequestQueue.Entry> entries = new ArrayList<>(Arrays.asList(testEntry1, testEntry2, testEntry3));
+
+        mockCompleteTxWithThrowForBuildBatchedPegout(wallet, entries, new InsufficientMoneyException(Coin.valueOf(1234)));
+
+        Optional<ReleaseTransactionBuilder.BuildResult> result = builder.buildBatchedPegouts(entries);
+
+        Assert.assertFalse(result.isPresent());
+        verify(wallet, never()).getWatchedAddresses();
+        verify(wallet, never()).getUTXOProvider();
+    }
+
```

Comment:
Seems we are missing tests for the other type of exceptions this method can throw (ExceededMaxTransactionSize, CouldNotAdjustDownwards, etc)

Reply:
Resolved